04_plot_stratified_results.ipynb

```diff
@@ -465,8 +465,8 @@
    ],
    "source": [
     "vogelstein_results_df = au.compare_results(vogelstein_df, metric='aupr', correction=True,\n",
-    "                                correction_method='fdr_bh', correction_alpha=0.001,\n",
-    "                                verbose=True)\n",
+    "                                           correction_method='fdr_bh', correction_alpha=0.001,\n",
```

Comment:
These are all good questions! See answers below:

> 1. I tend to find it helpful to have a description of the experiment I am performing at the top of the notebook just to orient myself.

Good idea! I'll add this to the top here.

> 2. Just wanted to clarify the term stratified here. So you're saying that you're training set includes say 10 samples of cancer type A, 10 of cancer type B, 10 of cancer type C (total of 30 samples each with 1/3 protion). And so your test set contains maybe 9 total samples with 3 samples of cancer type A, 3 of cancer type B, 3 of cancer type C. So the proportions are the same in the test and training..?

Yep, exactly (not always exactly the same proportion between train/test sets, but +/- 1 sample I think - this is implemented in [StratifiedKFold from scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html)). 

> 3. Trying to understand your dfs. So for the first 3 rows top50_df you have auroc and aupr that tells you how well gene info from training set (including mutation burden, etc) predict mutation status of TP53 (binary i assume) on training/test/validation sets where the labels used to train the model were shuffled. I assume that means you have the same training dataset with multiple labels for mutation status of gene X, Y, Z. So you'd train your model on its ability to predict mutation of gene X, then train your model on its ability to predict mutation of gene Y,...So you have multiple models here?

Right - so each row in `top50_df` and `vogelstein_df` is one model, trained on the (binary, mutated or not mutated) mutation status of one gene on one cross-validation fold, either on the true labels or the shuffled labels.

We have mutation information for (almost) every gene in the genome - "top_50" and "vogelstein" are two different ways to select the genes to train models on. If we just train models on every gene in the genome, our statistical power to detect true relationships between mutation and gene expression won't be very good (and also it will take forever), so we want to start with sets of known cancer genes to improve power. In each case, we train one model for each gene/true-shuffled/cross-validation fold combination.

Let me know if that doesn't answer your question.

Reply:
This makes sense. Thank you! 

As a followup. So you're including those cancer genes instead of all genes. Do you expect most of those genes to have a mutated status = 1 (I guess this'll probably depend on the cancer type)? Would it make sense to include genes that are not mutated in cancers as a control?

Reply:
>  Do you expect most of those genes to have a mutated status = 1 (I guess this'll probably depend on the cancer type)?

Yeah, we definitely expect many of the cancer-related genes to be frequently mutated in at least a few cancer types (and we are filtering out gene/cancer type combos where less than 5% of samples are mutated, like in my last exploratory data analysis notebook).

> Would it make sense to include genes that are not mutated in cancers as a control?

Yes, this is definitely what we're trying to do, but it's hard to choose control genes well. Ideally we'd include some genes that aren't drivers, but this isn't really documented anywhere (and absence of evidence for a gene being a driver in some cancer type isn't the same as evidence of absence; some drivers are just rarely mutated or haven't been studied in depth).

In the past we've used TTN as an example of a gene that isn't thought to be a driver of any cancer type, but remember that lots of genes are mutated in cancer, even those that aren't actually driving the cancer to form. TTN is a large gene that is frequently mutated as a passenger (just by chance) in many cancers, so its mutation status correlates with mutation burden (and thus with cancer type). So (we think) it turns out that TTN mutation status can actually be predicted to some degree from gene expression, because gene expression -> cancer type -> mutation burden -> probability of TTN being mutated.

I guess we could pick smaller genes to reduce the chances of passenger mutations, but I'd have to think about what the best way is to do this.